### PR TITLE
executor: migrate test-infra to testify for sort_test

### LINKED ISSUE #28616: migrate test-infra to testify for `sort_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28616

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/sort_test.go [hunk 1]
```diff
@@ -119,10 +122,10 @@ func (s *testSerialSuite1) TestIssue16696(c *C) {
 		line := fmt.Sprintf("%v", row)
 		disk := fmt.Sprintf("%v", row[length-1])
 		if strings.Contains(line, "Sort") || strings.Contains(line, "HashJoin") {
-			c.Assert(strings.Contains(disk, "0 Bytes"), IsFalse)
-			c.Assert(strings.Contains(disk, "MB") ||
+			require.False(t, strings.Contains(disk, "0 Bytes"))
+			require.True(t, strings.Contains(disk, "MB") ||
```
Comment (tisonkun) [inline executor/sort_test.go:126]: IIRC there is a `require.(Not)Contains` method to use.

Comment (hawkingrei) [reply executor/sort_test.go]: You are right. Fixed.

File: executor/sort_test.go [hunk 2]
```diff
@@ -119,10 +124,11 @@ func (s *testSerialSuite1) TestIssue16696(c *C) {
 		line := fmt.Sprintf("%v", row)
 		disk := fmt.Sprintf("%v", row[length-1])
 		if strings.Contains(line, "Sort") || strings.Contains(line, "HashJoin") {
-			c.Assert(strings.Contains(disk, "0 Bytes"), IsFalse)
-			c.Assert(strings.Contains(disk, "MB") ||
+			require.False(t, strings.Contains(disk, "0 Bytes"))
```
Comment (tisonkun) [inline executor/sort_test.go:127]:
```suggestion
```

**Review by tisonkun — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**